src/components/AgenciesPage/components/AgenciesList.tsx

```diff
@@ -1,14 +1,31 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Grid } from '@mui/material';
+import { useDispatch } from 'react-redux';
+import { ThunkDispatch } from '@reduxjs/toolkit';
 import { AgenciesItem } from './components';
-import { agenciesData } from '../mocks/agenciesData';
-
-export const AgenciesList = () => (
-  <Grid container>
-    {agenciesData.map(agency => (
-      <Grid item key={agency.id} xs={12} sm={6} md={4} lg={3}>
-        <AgenciesItem agency={agency} />
-      </Grid>
-    ))}
-  </Grid>
-);
+import { selectAgencies } from '../selectors/selectors';
+import { getAgenciesData } from '../services/getAgenciesData';
+import { AgenciesAction } from '../types/Agencies';
+import { StoreState } from '../../../store';
+
+export const AgenciesList = () => {
+  const dispatch =
+    useDispatch<ThunkDispatch<StoreState, void, AgenciesAction>>();
```

Comment:
If iam right first generic into ThunkDispatch must be what is return a function second state third extra arguments and next actions so first i think need be Promise<void>

Reply:
Sorry its about ThunkAction , ThunkDispatch really take 3 arguments right like in your case